File: 0219. Contains Duplicate II/Solution0219.py
class Solution0219:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        numset = set()
        for i,n in enumerate(nums):
            if n in numset: 
                return True
            numset.add(n)
            if len(numset) > k:
                numset.remove(nums[i-k])
        return False

File: 0219. Contains Duplicate II/Solution0219.cpp
class Solution0219 {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        set<int> numset;
        for (int i = 0; i < nums.size(); i++) {
            if (numset.find(nums[i]) != numset.end()) {
                return true;
            }
            numset.insert(nums[i]);
            if (numset.size() > k) {
                numset.erase(nums[i-k]);
            }
        }
        return false;
    }
};